src/ngSanitize/sanitize.js

```diff
@@ -518,8 +524,17 @@ function $SanitizeProvider() {
         stripCustomNsAttrs(nextNode);
       }
 
-      node = node.nextSibling;
+      node = getNonDescendant('nextSibling', node);
+    }
+  }
+
+  function getNonDescendant(propName, node) {
+    // An element is clobbered if its `propName` property points to one of its descendants
+    var nextNode = node[propName];
+    if (nextNode && nodeContains.call(node, nextNode)) {
```

Comment:
are we bothered about generic descendents or only inputs that are members of a form?
if the latter then could we do:

```
if (nextNode && nextNode.form === node)
```